### ISSUE TITLE
Unexpected state in beforeSpecHandler

### ISSUE THREAD

**Issue by ejbartelds** [badeball/cypress-cucumber-preprocessor#1046]
### Current behavior

While running Cypress tests:

An error was thrown in your plugins file while executing the handler for the before:spec event.

The error we received was:

Error: Unexpected state in beforeSpecHandler: before-spec (this might be a bug, please report at https://github.com/badeball/cypress-cucumber-preprocessor)
    at createError (/**/frontend/node_modules/@badeball/cypress-cucumber-preprocessor/dist/helpers/error.js:9:12)
    at beforeSpecHandler (/**/frontend/node_modules/@badeball/cypress-cucumber-preprocessor/dist/plugin-event-handlers.js:189:43)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)

### Desired behavior

No exceptions and unexpected state, as before

### Test code to reproduce

Working on it...

### Versions

* **Cypress version**: 12.15.0
* **Preprocessor version**: 18.0.1
* **Node version**: 18.16.1

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#bug-reports).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by MadSandwich** [comment]
I have the same issue. This is because `cypress-cloud` package (at least for me), without this package this issue is not reproduced. Looks like some incapability. @ejbartelds Do u have this package as well?

**Comment by badeball** [comment]
@MadSandwich, https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/event-handlers.md

**Comment by MadSandwich** [comment]
@badeball thank for help, I was able to solve problem.

**Comment by badeball** [comment]
Closing due to lack of a reproducible example. Feel free to open up another issue if anything changes in this regard.

**Comment by muskaanshrmaa** [comment]
For me this happened due to the version issue in badeball. I reverted back to version @badeball/cypress-cucumber-preprocessor@18.0.4 and it worked fine

**Comment by nealio42** [comment]
I can reproduce it with ... as @muskaanshrmaa has suggested, reverting back to 18.04  works.

```
on(
    'after:spec',
     (spec, results) => {
      if (results && results.video) {
        // Do we have failures for any retry attempts?
        const failures = results.tests.some((test) =>
          test.attempts.some((attempt) => attempt.state === 'failed')
        );
        if (!failures) {
          // delete the video if the spec passed and no tests retried
          fs.unlink(results.video, (err) => {
            if (err) console.log(err);
            else {
              console.log(`\nFor ${spec.name}, deleted file: ${results.video}`);
            }
          });
        }        
      }
    }
  );
```

**Comment by badeball** [comment]
@nealio42, are you trying to solicit help?

**Comment by nealio42** [comment]
Yes, but probably going about it incorrectly. I ran into this issue and
it’s kicking my butt. Going to try cypress-on-fix next. Maybe there is an
alternative way to remove the cypress video files as the example provided
by cypress causes this same error. For now, I will just leave the videos
pass or fail. Just wanted to use this plugin with after:spec, where
after:spec I can delete the videos for a passing spec.

**Comment by badeball** [comment]
`cypress-on-fix` is the answer, as has already been pointed out earlier.

**Comment by nealio42** [comment]
Ok, thanks for the clarification.

**Comment by nealio42** [comment]
Yes, this fixed my issue. Thank you for your direction.
